honeybee_energy/schedule/fixedinterval.py

```diff
@@ -0,0 +1,690 @@
+                'ScheduleTypeLimit for ScheduleRuleset schedule_type_limit. ' \
+                'Got {}.'.format(type(schedule_type))
+        self._schedule_type_limit = schedule_type
+
+    @property
+    def out_of_range_value(self):
+        """Get or set the value to be used for all times not covered by the input values.
+        """
+        return self._out_of_range_value
+
+    @out_of_range_value.setter
+    def out_of_range_value(self, value):
+        self._out_of_range_value = float_in_range(
+            value, input_name='schedule fixed interval out_of_range_value')
+
+    @property
+    def interpolate(self):
+        """Get or set a boolean noting whether values should be interpolated."""
+        return self._interpolate
+
+    @interpolate.setter
+    def interpolate(self, interpolate):
+        self._interpolate = bool(interpolate)
+
+    @property
+    def timestep(self):
+        """Get the integer for the schedule's number of steps per hour."""
+        return self._timestep
+
+    @property
+    def start_date(self):
+        """Get the ladybug Date object noting when the the input values take effect."""
+        return self._start_date
+
+    @property
+    def end_date_time(self):
+        """Get a ladybug DateTime object for the end time of the schedule's values."""
+        num_hoys = (len(self._values) - 1) / self.timestep
+        end_hoy = (self.start_date.doy - 1) * 24 + num_hoys
+        if not self.is_leap_year:
+            end_dt = DateTime.from_hoy(end_hoy) if end_hoy < 8760 else \
+                DateTime.from_hoy(end_hoy - 8760)
+        else:
+            end_dt = DateTime.from_hoy(end_hoy, True) if end_hoy < 8784 else \
+                DateTime.from_hoy(end_hoy - 8760, True)
+        return end_dt
+
+    @property
+    def is_leap_year(self):
+        """Get a boolean noting whether the schedule is over a leap year.
+
+        Note that this property originates from the leap_year property on the
+        input start_date.
+        """
+        return self._start_date.leap_year
+
+    @property
+    def data_collection(self):
+        """A DataCollection of schedule values at the native start_date and timestep."""
+        end_dt = self.end_date_time
+        a_period = AnalysisPeriod(self.start_date.month, self.start_date.day, 0,
+                                  end_dt.month, end_dt.day, end_dt.hour, self.timestep,
+                                  self.is_leap_year)
+        data_type, unit = self._get_lb_data_type_and_unit()
+        header = Header(data_type, unit, a_period, metadata={'schedule': self.name})
+        return HourlyContinuousCollection(header, self._values)
+
+    def values_at_timestep(
+            self, timestep=1, start_date=None, end_date=None):
+        """Get a list of sequential schedule values over the year at a given timestep.
+
+        Note that there are two possible ways that these values can be mapped to
```

Comment:
I would at least name the two here. And if possible mention how does it make a difference for the developer. Is it an input to switch between the two? Is there one that is already in use? The description right now just makes me less confident to use this method.

Reply:
Ok. I added a more detailed description according to your suggestion.